core/frontend/src/types/kraken.ts

```diff
@@ -0,0 +1,26 @@
+import { Dictionary } from '@/types/common'
+
+export interface Author {
+    name: string
+    email: string
+    about?: string
+}
+
+export interface Version {
+    permissions: any
+    requirements: string | null
+    tag: string
+}
+
+export interface ExtensionData {
+    name: string
```

Comment:
Besides name we should have a string or a tag the could be used to track the extension itself. 
The reason behind that is that the extension could change names between versions, or information as website, authors and docker may change for each version.
Maybe something like:
```typescript
export interface Extension {
    identifier: string // format: extension_owner_name.owner_extension_name, E.g: patrickelectric.tainhacpp
    data: ExtensionData[]
}

export interface ExtensionData {
    name: string
    description: string
    docker: string
    website: string
    version: Version
    authors?: Author[]
    readme?: string
    support: string
    docs?: string
    icon?: string
}
```
We could also break this in a list of Extension per version, but that would make things ugly (Like Arduino does), and do the proper filter and menu creation based on the json processing.

Reply:
I did it as we discussed in the call, and sneaked the identifier in there, too.